daft/dataframe/dataframe.py

```diff
@@ -233,6 +233,47 @@ def schema(self) -> Schema:
         """
         return self.__builder.schema()
 
+    @DataframePublicAPI
+    def describe(self, columns: ManyColumnsInputType = []) -> "DataFrame":
+        """Returns column statistics.
```

Comment:
[Discussed on Slack](https://dist-data.slack.com/archives/C052CA6Q9N1/p1737486557069129) and moving to "summarize" so that we can reserve "describe" for SQL consistency.

My only other feedback is to pivot the result table so the output is more like,

```
| column | type | nulls | approx_distinct | min | max |
|-----------------------------------------------------|
| a | int64 | ... |
| b | int64  | ... |
```

Which enables easier programmatic access to the statistics since the column names are fixed

Reply:
I guess there's no harm in keeping a `columns: List[str]` argument as it's probably quite nice to reduce the compute required.